packages/swapper/src/swappers/cow/utils/helpers/helpers.ts

```diff
@@ -0,0 +1,61 @@
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { Asset } from '@shapeshiftoss/types'
+import { AxiosResponse } from 'axios'
+
+import { SwapError, SwapErrorTypes } from '../../../../api'
+import { bn, bnOrZero } from '../../../utils/bignumber'
+import { CowSwapperDeps } from '../../CowSwapper'
+import { CowSwapPriceResponse } from '../../types'
+import { cowService } from '../cowService'
+
+const USDC_CONTRACT_ADDRESS = '0xa0b86991c6218b36c1d19d4a2e9eb0ce3606eb48'
+const USDC_ASSET_PRECISION = 6
+
+export const getUsdRate = async (
+  { assetService, apiUrl }: CowSwapperDeps,
+  input: Pick<Asset, 'assetId'>
+): Promise<string> => {
+  const { assetId } = input
+
+  const { assetReference: erc20Address, assetNamespace } = fromAssetId(assetId)
+
+  if (assetNamespace !== 'erc20') {
+    throw new SwapError('[getUsdRate] - unsupported asset namespace', {
+      code: SwapErrorTypes.USD_RATE_FAILED,
+      details: { assetNamespace }
+    })
+  }
+
+  const assets = assetService.getAll()
```

Comment:
if this is why we added the asset service dependency, don't do it this way.

in the near term future, the client will be aware of more assets than are statically defined in the service.

just get the assets from the store in the client (which is the source of truth) and pass that into the swapper instead

Reply:
Fixed this by adding `assets` dependency to swapper